src/main/java/org/candlepin/subscriptions/resource/OptInResource.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2009 - 2019 Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.subscriptions.resource;
+
+import org.candlepin.subscriptions.controller.OptInController;
+import org.candlepin.subscriptions.security.auth.AdminOnly;
+import org.candlepin.subscriptions.utilization.api.model.OptInConfig;
+import org.candlepin.subscriptions.utilization.api.resources.OptInApi;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+import javax.ws.rs.BadRequestException;
+
+/**
+ * Defines the API for opting an account and its org into the sync/reporting
+ * functionality.
+ */
+@Component
+@Profile("api")
+public class OptInResource implements OptInApi {
+
+    private OptInController controller;
+
+    @Autowired
+    public OptInResource(OptInController controller) {
+        this.controller = controller;
+    }
+
+    @AdminOnly
+    @Override
+    public void deleteOptInConfig() {
+        controller.optOut(validateAccountNumber(), validateOrgId());
+    }
+
+    @AdminOnly
+    @Override
+    public OptInConfig getOptInConfig() {
+        return controller.getOptInConfig(validateAccountNumber(), validateOrgId());
+    }
+
+    @AdminOnly
+    @Override
+    public OptInConfig putOptInConfig(Boolean enableTallySync, Boolean enableTallyReporting,
+        Boolean enableConduitSync) {
+        // NOTE: All query params are defaulted to 'true' by the API definition.
```

Comment:
Have we verified this is actually the case?  I like OpenAPI and all, but sometimes I've had it not do what I expected. :sob:

Reply:
> Have we verified this is actually the case? I like OpenAPI and all, but sometimes I've had it not do what I expected.

Yes, I've verified it in my manual testing. However, I'm missing a unit test that will confirm it. Thanks for pointing this out. Based on the empty unit test I noticed, apparently I was going to add more tests there, and forgot to. :/ I'll add them to verify that comment via a test.

Reply:
I decided to also attempt to default them in the resource class just in case something ever changed with openapi. Thanks for the feedback.

Reply:
DONE